src/os/bluestore/BlueStore.cc

```diff
@@ -12999,37 +13013,41 @@ int BlueStore::_clone(TransContext *txc,
 		   newo->onode.is_pgmeta_omap() ? PREFIX_PGMETA_OMAP
 		   : PREFIX_OMAP,
 		   newo->onode.nid);
+    newo->onode.clear_omap_flag();
   }
   if (oldo->onode.has_omap()) {
+    const string& prefix =
+      newo->onode.is_pgmeta_omap() ? PREFIX_PGMETA_OMAP : PREFIX_OMAP;
     dout(20) << __func__ << " copying omap data" << dendl;
     if (!newo->onode.has_omap()) {
       newo->onode.set_omap_flag();
       if (newo->oid.is_pgmeta()) {
 	newo->onode.flags |= bluestore_onode_t::FLAG_PGMETA_OMAP;
       }
+      string tail;
+      bufferlist tail_value;
+      get_omap_tail(newo->onode.nid, &tail);
+      txc->t->set(prefix, tail, tail_value);
     }
-    const string& prefix =
-      newo->onode.is_pgmeta_omap() ? PREFIX_PGMETA_OMAP : PREFIX_OMAP;
     KeyValueDB::Iterator it = db->get_iterator(prefix);
     string head, tail;
     get_omap_header(oldo->onode.nid, &head);
     get_omap_tail(oldo->onode.nid, &tail);
     it->lower_bound(head);
     while (it->valid()) {
-      if (it->key() >= tail) {
-	dout(30) << __func__ << "  reached tail" << dendl;
-	break;
-      } else {
-	dout(30) << __func__ << "  got header/data "
+      if (it->key() <= tail) {
```

Comment:
Shouldn't this be it->key() < tail to avoid duplicate tail set - we've just set tail at line 13030

Reply:
fixed